exchanges/order/order_test.go

```diff
@@ -2138,11 +1677,93 @@ func TestSideUnmarshal(t *testing.T) {
 	assert.ErrorAs(t, s.UnmarshalJSON([]byte(`14`)), &jErr, "non-string valid json is rejected")
 }
 
+func TestIsValid(t *testing.T) {
+	t.Parallel()
+	var timeInForceValidityMap = map[TimeInForce]bool{
+		TimeInForce(1):    false,
+		ImmediateOrCancel: true,
+		GoodTillTime:      true,
+		GoodTillCancel:    true,
+		GoodTillDay:       true,
+		FillOrKill:        true,
+		PostOnly:          true,
+		UnsetTIF:          true,
+		UnknownTIF:        false,
+	}
+	var tif TimeInForce
+	for tif = range timeInForceValidityMap {
+		assert.Equalf(t, timeInForceValidityMap[tif], tif.IsValid(), "got %v, expected %v for %v with id %d", tif.IsValid(), timeInForceValidityMap[tif], tif, tif)
+	}
+}
+
+func TestStringToTimeInForce(t *testing.T) {
+	t.Parallel()
+
+	var timeInForceStringToValueMap = map[string]struct {
+		TIF   TimeInForce
+		Error error
+	}{
+		"Unknown":                      {TIF: UnknownTIF, Error: ErrInvalidTimeInForce},
+		"GoodTillCancel":               {TIF: GoodTillCancel},
+		"GOOD_TILL_CANCELED":           {TIF: GoodTillCancel},
+		"GTT":                          {TIF: GoodTillTime},
+		"GOOD_TIL_TIME":                {TIF: GoodTillTime},
+		"FILLORKILL":                   {TIF: FillOrKill},
+		"POST_ONLY_GOOD_TIL_CANCELLED": {TIF: PostOnly},
+		"immedIate_Or_Cancel":          {TIF: ImmediateOrCancel},
+		"":                             {TIF: UnsetTIF},
+		"IOC":                          {TIF: ImmediateOrCancel},
+		"immediate_or_cancel":          {TIF: ImmediateOrCancel},
+		"IMMEDIATE_OR_CANCEL":          {TIF: ImmediateOrCancel},
+		"IMMEDIATEORCANCEL":            {TIF: ImmediateOrCancel},
+		"GOOD_TILL_CANCELLED":          {TIF: GoodTillCancel},
+		"good_till_day":                {TIF: GoodTillDay},
+		"GOOD_TILL_DAY":                {TIF: GoodTillDay},
+		"GTD":                          {TIF: GoodTillDay},
+		"GOODtillday":                  {TIF: GoodTillDay},
+		"abcdfeg":                      {TIF: UnknownTIF, Error: ErrInvalidTimeInForce},
+		"PoC":                          {TIF: ImmediateOrCancel},
+		"PendingORCANCEL":              {TIF: ImmediateOrCancel},
```

Comment:
Pending Or Cancel is a Post Only which allows a failure if it will get matched instantly (from wording). But in essence it won't match and it will force maker fees. See "POC" in gateio.